src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -22,428 +22,693 @@ namespace NuGet.Commands
+                    buildCrossTargetsGroup.RootName = MSBuildRestoreItemGroup.ImportGroup;
+                    buildCrossTargetsGroup.Position = 0;
+
+                    buildCrossTargetsGroup.Items.AddRange(sortedPackages.SelectMany(pkg =>
+                        pkg.Key.BuildMultiTargeting.WithExtension(TargetsExtension)
+                            .Where(e => pkg.Value.Exists())
+                            .Select(e => pkg.Value.GetAbsolutePath(e)))
+                            .Select(path => GetImportPath(path, repositoryRoot))
+                            .Select(GenerateImport));
+
+                    targets.AddRange(GetGroupsWithConditions(buildCrossTargetsGroup, isMultiTargeting, CrossTargetingCondition));
+
+                    // buildMultiTargeting/ {packageId}.props
+                    var buildCrossPropsGroup = new MSBuildRestoreItemGroup();
+                    buildCrossPropsGroup.RootName = MSBuildRestoreItemGroup.ImportGroup;
+                    buildCrossPropsGroup.Position = 0;
+
+                    buildCrossPropsGroup.Items.AddRange(sortedPackages.SelectMany(pkg =>
+                        pkg.Key.BuildMultiTargeting.WithExtension(PropsExtension)
+                            .Where(e => pkg.Value.Exists())
+                            .Select(e => pkg.Value.GetAbsolutePath(e)))
+                            .Select(path => GetImportPath(path, repositoryRoot))
+                            .Select(GenerateImport));
+
+                    props.AddRange(GetGroupsWithConditions(buildCrossPropsGroup, isMultiTargeting, CrossTargetingCondition));
                 }
 
-                // Skip libraries that do not include build files such as transitive packages
-                if (includeLibrary)
+                // ContentFiles are read by the build task, not by NuGet
+                // for UAP with project.json.
+                if (request.RestoreOutputType != RestoreOutputType.UAP)
                 {
-                    var packageIdentity = new PackageIdentity(library.Id, library.Version);
-                    IList<string> packageFiles;
-                    context.PackageFileCache.TryGetValue(packageIdentity, out packageFiles);
+                    // Create a group for every package, with the nearest from each of allLanguages
+                    props.AddRange(sortedPackages.Select(pkg =>
+                         pkg.Key.ContentFiles
+                                .Where(e => pkg.Value.Exists())
+                                .OrderBy(e => e.Path, StringComparer.Ordinal)
+                                .Select(e =>
+                                    new Tuple<LockFileTargetLibrary, LockFileContentFile, string>(
+                                        item1: pkg.Key,
+                                        item2: e,
+                                        item3: pkg.Value.GetAbsolutePath(GetImportPath(e.Path, repositoryRoot)))))
```

Comment:
Fixed in the latest commit. I have no clue how this was compiling even, GetAbsolutePath does not take a string...